FWIO/RNTuple/plugins/RNTupleOutputFile.cc

```diff
@@ -0,0 +1,583 @@
+      }
+      {
+        auto field = ROOT::Experimental::RFieldBase::Create(kEventSelName, "std::vector<edm::Hash<1> >").Unwrap();
+        model->AddField(std::move(field));
+      }
+      {
+        auto field = ROOT::Experimental::RFieldBase::Create(kBranchListName, "std::vector<unsigned short>").Unwrap();
+        model->AddField(std::move(field));
+      }
+      setupDataProducts(iProducts, iConfig.streamerProduct, iConfig.doNotSplitSubFields, *model);
+
+      auto writeOptions = ROOT::Experimental::RNTupleWriteOptions();
+      writeOptions.SetCompression(convert(iConfig.compressionAlgo), iConfig.compressionLevel);
+      writeOptions.SetApproxZippedClusterSize(iConfig.approxZippedClusterSize);
+      writeOptions.SetMaxUnzippedClusterSize(iConfig.maxUnzippedClusterSize);
+      writeOptions.SetInitialUnzippedPageSize(iConfig.initialUnzippedPageSize);
+      writeOptions.SetMaxUnzippedPageSize(iConfig.maxUnzippedPageSize);
+      writeOptions.SetPageBufferBudget(iConfig.pageBufferBudget);
+      writeOptions.SetUseBufferedWrite(iConfig.useBufferedWrite);
+      writeOptions.SetUseDirectIO(iConfig.useDirectIO);
+      events_ = ROOT::Experimental::RNTupleWriter::Append(std::move(model), "Events", file_, writeOptions);
+    }
+    products_[InEvent] = associateDataProducts(iProducts, events_->GetModel());
+
+    eventAuxField_ = events_->GetModel().GetToken(kEventAuxName);
+    eventProvField_ = events_->GetModel().GetToken(kEventProvName);
+    eventSelField_ = events_->GetModel().GetToken(kEventSelName);
+    branchListField_ = events_->GetModel().GetToken(kBranchListName);
+
+    // Note: The EventSelectionIDVector should have a one to one correspondence with the processes in the process history.
+    // Therefore, a new entry should be added if and only if the current process has been added to the process history,
+    // which is done if and only if there is a produced product.
+    extendSelectorConfig_ = anyProductProduced || !iConfig.wantAllEvents;
+  }
+  void RNTupleOutputFile::setupPSets(Config const& iConfig) {
+    auto model = ROOT::Experimental::RNTupleModel::CreateBare();
+    {
+      auto field = ROOT::Experimental::RFieldBase::Create("IdToParameterSetsBlobs",
+                                                          "std::pair<edm::Hash<1>,edm::ParameterSetBlob>")
+                       .Unwrap();
+      model->AddField(std::move(field));
+    }
+    auto writeOptions = ROOT::Experimental::RNTupleWriteOptions();
+    writeOptions.SetCompression(convert(iConfig.compressionAlgo), iConfig.compressionLevel);
+    parameterSets_ = ROOT::Experimental::RNTupleWriter::Append(std::move(model), "ParameterSets", file_, writeOptions);
+  }
+
+  void RNTupleOutputFile::fillPSets() {
+    std::pair<ParameterSetID, ParameterSetBlob> idToBlob;
+
+    auto rentry = parameterSets_->CreateEntry();
+    rentry->BindRawPtr("IdToParameterSetsBlobs", const_cast<void*>(static_cast<void const*>(&(idToBlob))));
```

Comment:
Why are the cast to `const` and then `const_cast` needed? Would
```suggestion
    rentry->BindRawPtr("IdToParameterSetsBlobs", static_cast<void*>(&(idToBlob)));
```
work?